### PR TITLE
change schedule related structs and vars naming for scheduler abstraction

### PR DESCRIPTION
**Pull request by shuox**
1. change naming   
    prepare_switch_out -> switch_out
    prepare_switch_in -> switch_in
    prepare_switch -> do_switch
    run_thread_t -> thread_t
    sched_object -> thread_object
    sched_object.thread -> thread_object.thread_entry
    sched_obj -> thread_obj
    sched_context -> sched_control
    sched_ctx -> sched_ctl
2. remove runqueue from current schedule logic
3. move pcpu_id from acrn_vcpu to thread_object

### CODE REVIEW

File: hypervisor/arch/x86/guest/vcpu.c [hunk 1]
```diff
@@ -668,7 +668,7 @@ void pause_vcpu(struct acrn_vcpu *vcpu, enum vcpu_state new_state)
 	vcpu->state = new_state;
 
 	if (vcpu->running) {
-		remove_from_cpu_runqueue(&vcpu->thread_obj);
+		remove_thread_obj(&vcpu->thread_obj, vcpu->pcpu_id);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vcpu.c]: remove_thread or remove_thread_obj?

Comment (dongyaozu) [reply hypervisor/arch/x86/guest/vcpu.c]: Which one is better

File: hypervisor/arch/x86/guest/vcpu.c [hunk 2]
```diff
@@ -697,7 +697,7 @@ void resume_vcpu(struct acrn_vcpu *vcpu)
 	vcpu->state = vcpu->prev_state;
 
 	if (vcpu->state == VCPU_RUNNING) {
-		add_to_cpu_runqueue(&vcpu->thread_obj, vcpu->pcpu_id);
+		insert_thread_obj(&vcpu->thread_obj, vcpu->pcpu_id);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vcpu.c]: ditto

File: hypervisor/common/schedule.c [hunk 3]
```diff
@@ -13,6 +13,14 @@
 #include <schedule.h>
 #include <sprintf.h>
 
+/**
+ * @pre obj != NULL
+ */
+uint16_t sched_get_pcpuid(const struct thread_object *obj)
+{
```
Comment (dongyaozu) [inline hypervisor/common/schedule.c]: get running or previously running PCPU ID of the thread?

Comment (shuox) [reply hypervisor/common/schedule.c]: get the id of PCPU which to run 'obj' .

**Comment by acrnsi** [comment]
OK to verify

**Pull request merged**